src/hardware/serialport/nullmodem.cpp

```diff
@@ -155,41 +150,53 @@ CNullModem::~CNullModem() {
 	if (serversocket) delete serversocket;
 	if (clientsocket) delete clientsocket;
 	// remove events
-	for(Bit16u i = SERIAL_BASE_EVENT_COUNT+1;
-			i <= SERIAL_NULLMODEM_EVENT_COUNT; i++) {
+	for (uint16_t i = SERIAL_BASE_EVENT_COUNT + 1;
+	     i <= SERIAL_NULLMODEM_EVENT_COUNT; i++) {
 		removeEvent(i);
 	}
 }
 
-void CNullModem::WriteChar(Bit8u data) {
-	if (clientsocket)clientsocket->SendByteBuffered(data);
+void CNullModem::WriteChar(uint8_t data)
+{
+	if (clientsocket)
+		clientsocket->SendByteBuffered(data);
 	if (!tx_block) {
-		//LOG_MSG("setevreduct");
+		// LOG_MSG("SERIAL: Port %" PRIu8 " setevreduct", GetPortNumber());
 		setEvent(SERIAL_TX_REDUCTION, (float)tx_gather);
 		tx_block=true;
 	}
 }
 
-Bits CNullModem::readChar() {
-	Bits rxchar = clientsocket->GetcharNonBlock();
-	if (telnet && rxchar>=0) return TelnetEmulation((Bit8u)rxchar);
-	else if (rxchar==0xff && !transparent) {// escape char
-		// get the next char
-		Bits rxchar = clientsocket->GetcharNonBlock();
-		if (rxchar==0xff) return rxchar; // 0xff 0xff -> 0xff was meant
-		rxchar&0x1? setCTS(true) : setCTS(false);
-		rxchar&0x2? setDSR(true) : setDSR(false);
-		if (rxchar&0x4) receiveByteEx(0x0,0x10);
-		return -1;	// no "payload" received
-	} else return rxchar;
+SocketState CNullModem::readChar(uint8_t &val)
```

Comment:
Replaces the mixed `>0, -1, -2` data/status `Bits` return code for actually populating a reference to a `uint8_t`, and returning a proper `enum SocketState` type.